src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs

```diff
@@ -80,16 +83,26 @@ protected override void SetOperations(OpenApiPathItem item)
                 {
                     if (LastSegmentIsKeySegment)
                     {
-                        UpdateRestrictionsType updateEntity = Context.Model.GetRecord<UpdateRestrictionsType>(_entityType);
-                        if (updateEntity?.IsUpdatable ?? true)
+                        UpdateRestrictionsType entityUpdateRestriction = Context.Model.GetRecord<UpdateRestrictionsType>(_navPropEntityType);
+                        UpdateRestrictionsType navPropUpdateRestriction = Context.Model.GetRecord<UpdateRestrictionsType>(NavigationProperty);
+                        if ((entityUpdateRestriction?.IsUpdatable ?? true) ||
+                            (navPropUpdateRestriction?.IsUpdatable ?? true))
```

Comment:
```suggestion
                        if ((entityUpdateRestriction?.IsUpdatable ?? true) &&
                            (navPropUpdateRestriction?.IsUpdatable ?? true))
```

Otherwise, if one even is false, it'll always be added in the absence of the other.

(replicate for the other operations types)